pkg/await/await.go

```diff
@@ -253,53 +166,24 @@ func Update(
 		return nil, err
 	}
 
-	// Wait until patch resolves as success or error.
+	// Wait until patch resolves as success or error. Note that the conditional is set up to log only
+	// if we don't have an entry for the resource type; in the event that we do, but the await logic
+	// is blank, simply do nothing instead of logging an error.
 	var waitErr error
 	id := fmt.Sprintf("%s/%s", currentSubmitted.GetAPIVersion(), currentSubmitted.GetKind())
-	switch id {
-	case appsV1Deployment, appsV1Beta1Deployment, appsV1Beta2Deployment, extensionsV1Beta1Deployment:
-		waitErr = untilAppsDeploymentUpdated(clientForResource, currentSubmitted)
-	case coreV1ReplicationController:
-		waitErr = untilCoreV1ReplicationControllerUpdated(clientForResource, currentSubmitted)
-	case coreV1ResourceQuota:
-		{
-			oldSpec, _ := pluck(lastSubmitted.Object, "spec")
-			newSpec, _ := pluck(currentSubmitted.Object, "spec")
-			if !reflect.DeepEqual(oldSpec, newSpec) {
-				waitErr = untilCoreV1ResourceQuotaUpdated(clientForResource, currentSubmitted)
+	if awaiter, exists := awaiters[id]; exists {
+		if awaiter.awaitUpdate != nil {
+			conf := updateAwaitConfig{
+				initAwaitConfig: initAwaitConfig{
+					pool: pool, disco: disco, clientForResource: clientForResource, currentInputs: currentSubmitted,
+				},
+				lastInputs:  lastSubmitted,
+				lastOutputs: liveOldObj,
 			}
+			waitErr = awaiter.awaitUpdate(conf)
 		}
-
-	// Cases where no wait is necessary.
-	case autoscalingV1HorizontalPodAutoscaler:
-	case storageV1StorageClass:
-	case coreV1ConfigMap:
-	case coreV1LimitRange:
-	case coreV1Namespace:
-	case coreV1PersistentVolume:
-	case coreV1PersistentVolumeClaim:
-	case coreV1Pod:
-	case coreV1Secret:
-	case coreV1Service:
-	case coreV1ServiceAccount:
-	case extensionsV1Beta1Ingress:
-	case rbacAuthorizationV1ClusterRole:
-	case rbacAuthorizationV1ClusterRoleBinding:
-	case rbacAuthorizationV1Role:
-	case rbacAuthorizationV1RoleBinding:
-	case rbacAuthorizationV1Alpha1ClusterRole:
-	case rbacAuthorizationV1Alpha1ClusterRoleBinding:
-	case rbacAuthorizationV1Alpha1Role:
-	case rbacAuthorizationV1Alpha1RoleBinding:
-	case rbacAuthorizationV1Beta1ClusterRole:
-	case rbacAuthorizationV1Beta1ClusterRoleBinding:
-	case rbacAuthorizationV1Beta1Role:
-	case rbacAuthorizationV1Beta1RoleBinding:
-		break
-
-	// TODO(hausdorff): Find some sensible default for unknown kinds.
-	default:
-		return nil, fmt.Errorf("Could not find object of type '%s'", id)
+	} else {
+		glog.V(1).Infof("No initialization logic found for object of type '%s'; defaulting to assuming initialization successful", id)
 	}
 
 	if waitErr != nil {
```

Comment:
Any reason not to move this return next to the single place where `waitErr` is now returned?  (And keep `waitErr` localized to that scope).